ui/webpack/devConfig.js

```diff
@@ -41,55 +63,70 @@ module.exports = {
         'memoizerific.js'
       ),
     ],
-    preLoaders: [
+    loaders: [
       {
         test: /\.js$/,
         exclude: /node_modules/,
         loader: 'eslint-loader',
-      },
-    ],
-    loaders: [
-      {
-        test: /\.json$/,
-        loader: 'json',
+        enforce: 'pre',
       },
       {
         test: /\.scss$/,
-        loader: ExtractTextPlugin.extract(
-          'style-loader',
-          'css-loader!sass-loader!resolve-url!sass?sourceMap'
-        ),
+        loader: ExtractTextPlugin.extract({
+          fallback: 'style-loader',
+          use: [
+            'css-loader',
+            'sass-loader',
+            'resolve-url-loader',
+            'sass-loader?sourceMap',
+          ],
+        }),
       },
       {
         test: /\.css$/,
-        loader: ExtractTextPlugin.extract(
-          'style-loader',
-          'css-loader!postcss-loader'
-        ),
+        loader: ExtractTextPlugin.extract({
+          fallback: 'style-loader',
+          use: ['css-loader', 'postcss-loader'],
+        }),
       },
       {
         test: /\.(ico|png|cur|jpg|ttf|eot|svg|woff(2)?)(\?[a-z0-9]+)?$/,
-        loader: 'file',
+        loader: 'file-loader',
       },
       {
         test: /\.js$/,
         exclude: /node_modules/,
-        loader: 'babel',
+        loader: 'babel-loader',
         query: {
           presets: ['es2015', 'react', 'stage-0'],
           cacheDirectory: true, // use a cache directory to speed up compilation
         },
       },
+      {
+        test: /\.ts(x?)$/,
+        exclude: /node_modules/,
+        use: [
+          babelLoader,
+          {
+            loader: 'ts-loader',
```

Comment:
At a recent meetup about TypeScript and React in Zagreb, it was suggested to use `awesome-typescript-loader` rather than `ts-loader` for a number of reasons I've now forgotten, but I'm curious if you've investigated the difference? A quick search seems to indicate that some projects find faster compile speeds with one or the other, but they seem to be generally equivalent. The creator of `awesome-typescript-loader` points out some differences here: https://github.com/s-panferov/awesome-typescript-loader/issues/94. It may just be something to keep in mind or experiment with once this is merged.